### ISSUE TITLE
Improving Python interface to verification and loss functions.

### ISSUE THREAD

**Issue by MatthewDaggitt** [vehicle-lang/vehicle#1191]
Okay, having spent the last week messing around with the Python interface in anger, I have a bunch of ideas of how to improve it.

Currently, we have:
- `vehicle_lang.verify`
- `vehicle_lang.loss.tensorflow`
- `vehicle_lang.loss.pytorch`

However, the users have to manually convert backwards and forwards from tensorflow/pytorch to ONNX if they want to use both the verify and loss backends. It feels like really we should have the modules:
- `vehicle_lang.tensorflow` which exports both loss generation and verification, with the latter performing the required conversions from tensorflow models to `.onnx` and from tensorflow tensors to `.idx` underneath the hood.
- `vehicle_lang.pytorch` similarly which would expose both. 
- `vehicle_lang.onnx` which would only export the verification interface.

This would allow users to drastically decrease the amount of boilerplate they have to write to get a complete pipeline setup.
